src/dataflow/src/source/kafka.rs

```diff
@@ -27,8 +27,21 @@ use kafka_util::KafkaAddrs;
 use log::{error, info, log_enabled, warn};
 use uuid::Uuid;
 
+use crate::logging::materialized::{Logger, MaterializedEvent};
 use crate::source::{NextMessage, SourceMessage, SourceReader};
 
+/// Values recorded from the last rdkafka statistics callback, used to generate a
+/// diff of values for logging
+pub struct PreviousStats {
```

Comment:
if you #[derive(Default)] you don't have to explicitly zero init later

Reply:
Nice! Looks like `Option` defines it's default as `None` as well, so this should work nicely.

Reply:
Done!